Internal review for the performance of VectorAPI on N2

e1iu opened the pull request.

Commit:
AArch64: [vectorapi] Optimize Vector.lane

This patch aims to optimize extract operation on vector for AArch64
according to Neoverse N2 and V1 software optimization guide[1][2].

Currently, extract operation is used by "Vector.lane"[3]. As SVE
doesn’t have direct instruction support for such operation like
"pextr"[4] in x86, the final code is as below:

```
	Byte512Vector.lane(7)

        orr     x8, xzr, #0x7
        whilele p0.b, xzr, x8
        lastb   w10, p0, z16.b
        sxtb    w10, w10
```

This patch uses NEON instruction instead if the target lane is located
in the NEON 128b range. For the same example above, the generated code
is much simpler:

```
        smov    x11, v16.b[7]
```

For those cases that target lane is located out of the NEON 128b range,
this patch uses EXT to shift the target to the lowest. The generated
code is as below:

```
        Byte512Vector.lane(63)

        mov     z17.d, z16.d
        ext     z17.b, z17.b, z17.b, #63
        smov    x10, v17.b[0]
```

hotspot/compiler/vectorapi, jdk/incubator/vector passed on SVE machine.

Refs:

 // From Arm Neoverse N2 and V1 Software Optimization Guide
 // NOTE: The data inside "()" belongs to V1
 +--------------+---------+------------+--------------------+
 |  Instruction | Latency | Throughput | Utilized Pipelines |
 +--------------+---------+------------+--------------------+
 |WHILELE       |        3|      1(1/2)|               M(M0)|
 +--------------+---------+------------+--------------------+
 |LASTB(scalar) |     5(6)|           1|               V1,M0|
 +--------------+---------+------------+--------------------+
 |EXT           |        2|           2|              V(V01)|
 +--------------+---------+------------+--------------------+
 |UMOV,SMOV     |        2|           1|                   V|
 +--------------+---------+------------+--------------------+
 |ORR           |        2|           2|              V(V01)|
 +--------------+---------+------------+--------------------+
 |INS           |        2|        2(4)|                   V|
 +--------------+---------+------------+--------------------+

[1] https://developer.arm.com/documentation/pjdoc466751330-9685/latest/
[2] https://developer.arm.com/documentation/PJDOC-466751330-18256/0001
[3] https://github.com/openjdk/jdk/blob/master/src/jdk.incubator.vector/share/classes/jdk/incubator/vector/IntVector.java#L2693
[4] https://www.felixcloutier.com/x86/pextrb:pextrd:pextrq

Change-Id: I90cfc1f8deb84145f42132d58d3b211c4a8933ad

Commit:
AArch64: [vectorapi] Optimize Vector.reduceLanes for 64 and 128 SPECIES

This patch speeds up add/mul/min/max reduction for 64 and 128 SPECIES
on N2 machine.

According to Neoverse N2 software optimization guide[1], ASIMD
reduction instructions are faster than SVE's for 128-bit vector size.
This patch adds some rules to distinguish 64 bits and 128 bits vector
size, so that for these two special cases, they can generate code the
same as NEON. E.g., For ByteVector.SPECIES_128,
"ByteVector.reduceLanes(VectorOperators.ADD)" generates code as below:

```
        Before:
        orr     x8, xzr, #0x10
        whilelo p0.b, xzr, x8
        uaddv   d17, p0, z16.b
        smov    x15, v17.b[0]
        add     w15, w14, w15, sxtb

        After:
        addv    b17, v16.16b
        smov    x12, v17.b[0]
        add     w12, w12, w16, sxtb
```

The performance improvements 60% ~ 100% on my test machine.

[1] https://developer.arm.com/documentation/PJDOC-466751330-18256/0001

Change-Id: Id4637cd4b0b7948864780eafa84150787697e4df

Commit:
AArch64: [vectorapi] Optimize Vector.withLane for SVE

This patch optimizes Vector.withLane for 64 and 128 SPECIES. For 64-
and 128-bit vector, insert operation could be implemented by ASIMD
instructions, for better performance. E.g., for IntVector.SPECIES_128,
"IntVector.withLane(0, (int)4)" generates code as below:

```
        Before:
        orr     w10, wzr, #0x4
        index   z17.s, #-16, #1
        cmpeq   p0.s, p7/z, z17.s, #-16
        mov     z17.d, z16.d
        mov     z17.s, p0/m, w10

        After
        orr     w10, wzr, #0x4
        mov     v16.s[0], w10
```

This patch also does a small enhancement for vector whose size is
greater than 128 bits, that it may save 1 "DUP" just if the target index
is smaller than 32. E.g., For ByteVector.SPECIES_512,
"ByteVector.withLane(0, (byte)4)" generates code as below:

```
        Before:
        index   z18.b, #0, #1
        mov     z17.b, #0
        cmpeq   p0.b, p7/z, z18.b, z17.b
        mov     z17.d, z16.d
        mov     z17.b, p0/m, w16

        After:
        index   z17.b, #-16, #1
        cmpeq   p0.b, p7/z, z17.b, #-16
        mov     z17.d, z16.d
        mov     z17.b, p0/m, w16
```

Change-Id: I700a28bc2fc15b6baca03b8d8574bb17992bf4a7

Commit:
8277795: ldap connection timeout not honoured under contention

Reviewed-by: dfuchs, aefimov

JesperIRL committed:
Merge

manukumarvs committed:
8280948: [TESTBUG] Write a regression test for JDK-4659800

Reviewed-by: aivanov

albertnetymk committed:
8281120: G1: Rename G1BlockOffsetTablePart::alloc_block to update_for_block

Reviewed-by: tschatzl, iwalulya, sjohanss

Commit:
8280476: [macOS] : hotspot arm64 bug exposed by latest clang

Reviewed-by: kbarrett, adinn

yminqi committed:
8280767: -XX:ArchiveClassesAtExit does not archive BoundMethodHandle$Species classes

Reviewed-by: iklam, ccheung

kevinjwalls committed:
8281049: man page update for jstatd Security Manager dependency removal

Reviewed-by: cjplummer

XueleiFan committed:
8281289: Improve with List.copyOf

Reviewed-by: jnimeh, hchao

jddarcy committed:
8281183: RandomGenerator:NextDouble() default behavior partially fixed by JDK-8280950

Reviewed-by: jlaskey

Commit:
8139173: [macosx] JInternalFrame shadow is not properly drawn

Reviewed-by: prr, serb

prrace committed:
8279878: java/awt/font/JNICheck/JNICheck.sh test fails on Ubuntu 21.10

Reviewed-by: serb

prsadhuk committed:
8166050: partialArray is not created in javax.swing.text.html.parser.NPrintWriter.println(...) method

Reviewed-by: prr

XueleiFan committed:
8281298: Revise the creation of unmodifiable list

Reviewed-by: redestad

eme64 committed:
8281117: Add regression test for JDK-8280587

Reviewed-by: chagedorn, thartmann, xliu

Commit:
8281243: Test java/lang/instrument/RetransformWithMethodParametersTest.java is failing

Reviewed-by: sspitsyn, dcubed, lmesnik

Commit:
8280965: Tests com/sun/net/httpserver/simpleserver fail with FileSystemException on Windows 11

Reviewed-by: dfuchs

AlekseiEfimov committed:
8272996: JNDI DNS provider fails to resolve SRV entries when IPV6 stack is enabled

Reviewed-by: dfuchs

albertnetymk committed:
8281114: G1: Remove PreservedMarks::init_forwarded_mark

Reviewed-by: tschatzl, sjohanss

Commit:
8268387: Rename maximum compaction to maximal compaction in G1

Reviewed-by: ayang, tschatzl

seanjmullan committed:
8280890: Cannot use '-Djava.system.class.loader' with class loader in signed JAR

Reviewed-by: weijun, hchao

wangweij committed:
8281175: Add a -providerPath option to jarsigner

Reviewed-by: xuelei, hchao

kevinjwalls committed:
8281377: Remove vmTestbase/nsk/monitoring/ThreadMXBean/ThreadInfo/Deadlock/JavaDeadlock001/TestDescription.java from problemlist.

Reviewed-by: sspitsyn

kevinjwalls committed:
6779701: Wrong defect ID in the code of test LocalRMIServerSocketFactoryTest.java

Reviewed-by: cjplummer, dfuchs

egahlin committed:
8279613: JFR: Snippify Javadoc

Reviewed-by: mgronlun

shipilev committed:
8281314: Rename Stack{Red,Yellow,Reserved,Shadow}Pages multipliers

Reviewed-by: stuefe, coleenp, xliu

Commit:
8279329: Remove hardcoded IPv4 available policy on Windows

Reviewed-by: djelinski, alanb, dfuchs, aefimov

Commit:
8280917: Simplify G1ConcurrentRefineThread activation

Reviewed-by: iwalulya, sjohanss

manukumarvs committed:
8281296: Create a regression test for JDK-4515999

Reviewed-by: aivanov

tstuefe committed:
8281450: Remove unnecessary operator new and delete from ObjectMonitor

Reviewed-by: dholmes

Commit:
8281400: Remove unused wcslen() function

Reviewed-by: dcubed, coleenp, lfoltan

TheRealMDoerr committed:
8281061: [s390] JFR runs into assertions while validating interpreter frames

Reviewed-by: lucy, rrich

sormuras committed:
8281104: jar --create should create missing parent directories

Reviewed-by: lancea

Commit:
8281476: ProblemList tools/jar/CreateMissingParentDirectories.java

Reviewed-by: azvegint, bpb, lancea

Commit:
8280828: Improve invariants in NonblockingQueue::append

Reviewed-by: iwalulya, tschatzl

merykitty committed:
8278947: Support for array constants in constant table

Reviewed-by: kvn, vlivanov

mrserb committed:
8280132: Incorrect comparator com.sun.beans.introspect.MethodInfo.MethodOrder

Reviewed-by: prr

Commit:
8280830: Change NonblockingQueue::try_pop variable named "result"

Reviewed-by: dholmes

Commit:
8280438: Improve BufferNode::Allocator::release to avoid walking pending list

Reviewed-by: iwalulya, tschatzl

shipilev committed:
8281168: Micro-optimize VarForm.getMemberName for interpreter

Reviewed-by: redestad, vlivanov, mchung

savoptik committed:
8281338: NSAccessibilityPressAction action for tree node and NSAccessibilityShowMenuAcgtion action not working

Reviewed-by: ant, kizune

Commit:
8272807: Permit use of memory concurrent with pretouch

Reviewed-by: shade, stuefe

djelinski committed:
8281440: AWT: Conversion from string literal loses const qualifier

Reviewed-by: prr, aivanov

Commit:
8281195: Mistakenly used logging causes significant overhead in interpreter

Reviewed-by: shade, dholmes

rwestrel committed:
8262721: Add Tests to verify single iteration loops are properly optimized

Reviewed-by: neliasso, chagedorn, kvn

MBaesken committed:
8281274: deal with ActiveProcessorCount in os::Linux::print_container_info

Reviewed-by: stuefe, sgehwolf, dholmes, iklam

sormuras committed:
8281470: tools/jar/CreateMissingParentDirectories.java fails with "Should have failed creating jar file"

Reviewed-by: lancea

Bhavana-Kilambi committed:
8280007: Enable Neoverse N1 optimizations for Arm Neoverse V1 & N2

Reviewed-by: phh

Commit:
8279822: CI: Constant pool entries in error state are not supported

Reviewed-by: kvn, thartmann

Commit:
8265765: DomainKeyStore may stop enumerating aliases if a constituting KeyStore is empty

Reviewed-by: weijun

aivanov-jdk committed:
8280820: Clean up bug8033699 and bug8075609.java tests: regtesthelpers aren't used

Reviewed-by: prr

Commit:
8281567: Remove @throws IOException from X509CRLImpl::getExtension docs

Reviewed-by: xuelei, jiefu

eme64 committed:
8252496: C2: Useless code in MergeMemNode::Ideal

Reviewed-by: thartmann, chagedorn, vlivanov

lkorinth committed:
8281379: Assign package declarations to all jtreg test cases under gc

Reviewed-by: kbarrett, tschatzl

mkartashev committed:
8281262: Windows builds in different directories are not fully reproducible

Co-authored-by: Erik Joelsson <erikj@openjdk.org>
Reviewed-by: erikj, ihse

Commit:
8280832: Update usage docs for NonblockingQueue

Reviewed-by: iwalulya, dholmes

prsadhuk committed:
8054449: Incompatible type in example code in TreePath

Reviewed-by: aivanov, dmarkov

Commit:
8281294: [vectorapi] FIRST_NONZERO reduction operation throws IllegalArgumentExcept on zero vectors

Reviewed-by: jrose

egahlin committed:
8281536: JFR: Improve jdk.jfr.ContentType documentation

Reviewed-by: mgronlun

Commit:
8281275: Upgrading from 8 to 11 no longer accepts '/' as filepath separator in gc paths

Reviewed-by: shade, dcubed

sandlerwang commented:
I have tested on Yitian 710 server.  The following tests got more than 10X improvement.
b.jdk.incubator.vector.DoubleMaxVector.MULLanes 31.6649525
b.jdk.incubator.vector.Double128Vector.MULLanes 31.23600078
b.jdk.incubator.vector.Double128Vector.MULMaskedLanes   29.12242118
b.jdk.incubator.vector.DoubleMaxVector.MULMaskedLanes   28.99145176
b.jdk.incubator.vector.Long128Vector.MULLanes   28.7557894
b.jdk.incubator.vector.LongMaxVector.MULLanes   28.44699175
b.jdk.incubator.vector.Long128Vector.MULMaskedLanes     28.14659362
b.jdk.incubator.vector.LongMaxVector.MULMaskedLanes     27.99170547
b.jdk.incubator.vector.Float64Vector.MULLanes   27.23323051
b.jdk.incubator.vector.Int64Vector.MULMaskedLanes       25.71397358
b.jdk.incubator.vector.Float64Vector.MULMaskedLanes     25.59713332
b.jdk.incubator.vector.Int64Vector.MULLanes     24.91029816
b.jdk.incubator.vector.Int128Vector.MULMaskedLanes      21.95332536
b.jdk.incubator.vector.Short64Vector.MULMaskedLanes     20.9384198
b.jdk.incubator.vector.Int128Vector.MULLanes    20.4989715
b.jdk.incubator.vector.IntMaxVector.MULLanes    20.10063869
b.jdk.incubator.vector.Float128Vector.MULMaskedLanes    19.6528013
b.jdk.incubator.vector.Short64Vector.MULLanes   18.5722387
b.jdk.incubator.vector.IntMaxVector.MULMaskedLanes      18.26829118
b.jdk.incubator.vector.ShortMaxVector.MULLanes  17.32804972
b.jdk.incubator.vector.Short128Vector.MULLanes  17.29941413
b.jdk.incubator.vector.FloatMaxVector.MULMaskedLanes    16.09921692
b.jdk.incubator.vector.Byte64Vector.MULMaskedLanes      15.88618623
b.jdk.incubator.vector.Byte128Vector.MULLanes   15.77287804
b.jdk.incubator.vector.Short128Vector.MULMaskedLanes    15.59241254
b.jdk.incubator.vector.ByteMaxVector.MULLanes   15.5877299
b.jdk.incubator.vector.ShortMaxVector.MULMaskedLanes    15.35547184
b.jdk.incubator.vector.FloatMaxVector.MULLanes  14.37003336
b.jdk.incubator.vector.Float128Vector.MULLanes  14.13716353
b.jdk.incubator.vector.Byte64Vector.MULLanes    13.93552747
b.jdk.incubator.vector.ByteMaxVector.MULMaskedLanes     13.66764554
b.jdk.incubator.vector.Byte128Vector.MULMaskedLanes     11.98839275

tstuefe committed:
8281522: Rename ADLC classes which have the same name as hotspot variants

Reviewed-by: neliasso, kvn

shipilev committed:
8281318: Improve jfr/event/allocation tests reliability

Reviewed-by: mgronlun

tstuefe committed:
8281460: Let ObjectMonitor have its own NMT category

Reviewed-by: dholmes, dcubed, shade

mrserb committed:
8281419: The source data for the color conversion can be discarded

Reviewed-by: prr, aivanov

shipilev committed:
8072070: Improve interpreter stack banging

Reviewed-by: xliu, coleenp, mdoerr

Commit:
8281626: NonblockingQueue should use nullptr

Reviewed-by: shade, dholmes

prsadhuk committed:
8047749: javadoc for getPathBounds() in TreeUI and BasicTreeUI is incorrect

Reviewed-by: aivanov

DamonFool committed:
8281638: jfr/event/allocation tests fail with release VMs after JDK-8281318 due to lack of -XX:+UnlockDiagnosticVMOptions

Reviewed-by: shade

lahodaj committed:
8281100: Spurious "variable might not have been initialized" with sealed class switch

Reviewed-by: vromero

lawrence-andrew committed:
8202836: [macosx] test java/awt/Graphics/TextAAHintsTest.java fails

Reviewed-by: prr, aivanov

djelinski committed:
8281259: MutableBigInteger subtraction could be simplified

Reviewed-by: bpb

yminqi committed:
8279997: check_for_dynamic_dump should not exit vm

Reviewed-by: ccheung, iklam

egahlin committed:
8281622: JFR: Improve documentation of jdk.jfr.Relational

Reviewed-by: jbachorik

calvinccheung committed:
8223077: module path support for dynamic CDS archive

Reviewed-by: iklam, minqi

manukumarvs committed:
8281535: Create a regression test for JDK-4670051

Reviewed-by: aivanov

kabutz committed:
8277175: Add a parallel multiply method to BigInteger

Reviewed-by: psandoz

jddarcy committed:
8281238: TYPE_USE annotations not printed in correct position in toString output

Reviewed-by: vromero

jddarcy committed:
8281462: Annotation toString output for enum not reusable for source input

Reviewed-by: mchung

jddarcy committed:
8281674: tools/javac/annotations/typeAnnotations/classfile/AnonymousExtendsTest.java fails with AssertionError

Reviewed-by: vromero

Commit:
8281033: Improve ImageCheckboxTest to test all available LaF

Reviewed-by: serb

Commit:
8274524: SSLSocket.close() hangs if it is called during the ssl handshake

Reviewed-by: xuelei

eme64 committed:
8278423: ExtendedDTraceProbes should be deprecated

Reviewed-by: dholmes, hseigel, kvn, thartmann

Commit:
8281675: VMDeprecatedOptions test fails after JDK-8278423

Reviewed-by: dcubed

mrserb committed:
8274939: Incorrect size of the pixel storage is used by the robot on macOS

Reviewed-by: aivanov, prr

Bhavana-Kilambi committed:
8239927: Product variable PrefetchFieldsAhead is unused and should be removed

Reviewed-by: njian, dholmes

theRealELiu commented:
> I have tested on Yitian 710 server. The following tests got more than 10X improvement. The number after the test name is speedup of this patch(score with patch/score without patch, the bigger the better) b.jdk.incubator.vector.DoubleMaxVector.MULLanes 31.6649525
b.jdk.incubator.vector.Double128Vector.MULLanes 31.23600078
b.jdk.incubator.vector.Double128Vector.MULMaskedLanes 29.12242118 b.jdk.incubator.vector.DoubleMaxVector.MULMaskedLanes 28.99145176
b.jdk.incubator.vector.Long128Vector.MULLanes 28.7557894
b.jdk.incubator.vector.LongMaxVector.MULLanes 28.44699175
b.jdk.incubator.vector.Long128Vector.MULMaskedLanes 28.14659362 b.jdk.incubator.vector.LongMaxVector.MULMaskedLanes 27.99170547
b.jdk.incubator.vector.Float64Vector.MULLanes 27.23323051
b.jdk.incubator.vector.Int64Vector.MULMaskedLanes 25.71397358
b.jdk.incubator.vector.Float64Vector.MULMaskedLanes 25.59713332
b.jdk.incubator.vector.Int64Vector.MULLanes 24.91029816
b.jdk.incubator.vector.Int128Vector.MULMaskedLanes 21.95332536
b.jdk.incubator.vector.Short64Vector.MULMaskedLanes 20.9384198
b.jdk.incubator.vector.Int128Vector.MULLanes 20.4989715
b.jdk.incubator.vector.IntMaxVector.MULLanes 20.10063869
b.jdk.incubator.vector.Float128Vector.MULMaskedLanes 19.6528013
b.jdk.incubator.vector.Short64Vector.MULLanes 18.5722387
b.jdk.incubator.vector.IntMaxVector.MULMaskedLanes 18.26829118
b.jdk.incubator.vector.ShortMaxVector.MULLanes 17.32804972
b.jdk.incubator.vector.Short128Vector.MULLanes 17.29941413
b.jdk.incubator.vector.FloatMaxVector.MULMaskedLanes 16.09921692
b.jdk.incubator.vector.Byte64Vector.MULMaskedLanes 15.88618623
b.jdk.incubator.vector.Byte128Vector.MULLanes 15.77287804
b.jdk.incubator.vector.Short128Vector.MULMaskedLanes 15.59241254
b.jdk.incubator.vector.ByteMaxVector.MULLanes 15.5877299
b.jdk.incubator.vector.ShortMaxVector.MULMaskedLanes 15.35547184
b.jdk.incubator.vector.FloatMaxVector.MULLanes 14.37003336
b.jdk.incubator.vector.Float128Vector.MULLanes 14.13716353
b.jdk.incubator.vector.Byte64Vector.MULLanes 13.93552747
b.jdk.incubator.vector.ByteMaxVector.MULMaskedLanes 13.66764554
b.jdk.incubator.vector.Byte128Vector.MULMaskedLanes 11.98839275

Thanks for your feedback. These improvements make more sense to me.  Do you find any improvement about `lane`, `withLane`, `MINLanes`, `MaxLanes`, and `ADDLanes` ? This patch used the low cost NEON instructions for these operations, giving by the N2 guide. I think they should have some performance changes.

> 
> While some cases got regression. The following cases got more than 10% regression b.jdk.incubator.vector.Long128Vector.ADDMaskedLanes 0.785240278 b.jdk.incubator.vector.LongMaxVector.ADDMaskedLanes 0.785526159
b.jdk.incubator.vector.Int128Vector.ADDMaskedLanes 0.81169551 
b.jdk.incubator.vector.IntMaxVector.ADDMaskedLanes 0.811933379
b.jdk.incubator.vector.Short128Vector.MAXMaskedLanes 0.839694945 b.jdk.incubator.vector.ShortMaxVector.MAXMaskedLanes 0.839843701 b.jdk.incubator.vector.Short128Vector.MINMaskedLanes 0.841166407 b.jdk.incubator.vector.ShortMaxVector.MINMaskedLanes 0.842104391 
b.jdk.incubator.vector.Int128Vector.MINMaskedLanes 0.866157601 
b.jdk.incubator.vector.IntMaxVector.MINMaskedLanes 0.867274704 
b.jdk.incubator.vector.Int128Vector.MAXMaskedLanes 0.874539806
b.jdk.incubator.vector.IntMaxVector.MAXMaskedLanes 0.874211224
b.jdk.incubator.vector.Short128Vector.ADDMaskedLanes 0.895701169 b.jdk.incubator.vector.ShortMaxVector.ADDMaskedLanes 0.896560222

These may related to the my patch. Since the NEON's reductions have low cost, so that even for those masked reduction operations, paying an extra VectorBlend maybe worth.

```
Before:

 ConI LoadVector   LoadVector
   |    /            |
 Binary      VectorLoadMask
   |           /
 AddReductionVI
B8:
  0x0000ffffb0488430:   ldr     w10, [x0, #12]
  0x0000ffffb0488434:   lsl     x10, x10, #3
  0x0000ffffb0488438:   ldr     s16, [x10, #16]              => LoadVector
  0x0000ffffb048843c:   sunpklo z17.h, z16.b
  0x0000ffffb0488440:   sunpklo z17.s, z17.h
  0x0000ffffb0488444:   cmpne   p0.s, p7/z, z17.s, #0        => VectorLoadMask
  0x0000ffffb0488448:   mov     w12, wzr
  0x0000ffffb048844c:   add     x9, sp, #0x20
  0x0000ffffb0488450:   ldr     z17, [x9]                    => LoadVector
 ;; 0x4
  0x0000ffffb0488454:   orr     x8, xzr, #0x4
  0x0000ffffb0488458:   whilelo p1.s, xzr, x8
  0x0000ffffb048845c:   mov     p1.b, p1/z, p0.b
  0x0000ffffb0488460:   uaddv   d16, p1, z17.s
  0x0000ffffb0488464:   smov    x10, v16.s[0]
  0x0000ffffb0488468:   add     w10, w10, w12                => AddReductionVI
  0x0000ffffb048846c:   ldr     w11, [x29, #12]   


After:

 ConI
  |   \
  |  ReplicateI     LoadVector      LoadVector
  |      |         /                 |
  |   Binary ------     VectorLoadMask
  |      |                /
  |   VectorBlend--------
  |      |
 AddReductionVI

  0x0000ffffa81787b4:   ldr     w10, [x0, #12]
  0x0000ffffa81787b8:   lsl     x10, x10, #3
  0x0000ffffa81787bc:   ldr     s16, [x10, #16]              => LoadVector
  0x0000ffffa81787c0:   sunpklo z17.h, z16.b
  0x0000ffffa81787c4:   sunpklo z17.s, z17.h
  0x0000ffffa81787c8:   cmpne   p0.s, p7/z, z17.s, #0        => VectorLoadMask
  0x0000ffffa81787cc:   ldr     z18, [sp]                    => spill [sp, #0] -> V18
  0x0000ffffa81787d0:   add     x9, sp, #0x60
  0x0000ffffa81787d4:   ldr     z16, [x9]                    => spill [sp, #96] -> V16
  0x0000ffffa81787d8:   sel     z16.s, p0, z16.s, z18.s      => VectorBlend
  0x0000ffffa81787dc:   mov     w11, wzr
 ;; neon_add_reduction_integral {
  0x0000ffffa81787e0:   addv    s17, v16.4s
  0x0000ffffa81787e4:   mov     w10, v17.s[0]
  0x0000ffffa81787e8:   add     w10, w10, w11
 ;; } neon_add_reduction_integral


```
In my local machine, it worth to add the extra VectorBlend, for using NEON's low cost `addv` in 128 bits cases, the benchmark looks good that performance improves doubled. But perhaps it's not a very general optimization that in some cases register spill happens, and the new added `ldr` perhaps the reason why it's not improved as expected but regression a little.  That needs more further investigation.

> b.jdk.incubator.vector.Double512Vector.broadcast 0.802555465 
b.jdk.incubator.vector.Float256Vector.MULMaskedLanes 0.811078434 
b.jdk.incubator.vector.Int256Vector.XORMaskedLanes 0.824115328 b.jdk.incubator.vector.Double256Vector.MAXMaskedLanes 0.824336765 
b.jdk.incubator.vector.Short256Vector.scatter 0.873017869 
b.jdk.incubator.vector.Long256Vector.MINMaskedLanes 0.873708785
b.jdk.incubator.vector.Int512Vector.MAXMaskedLanes 0.874315805 
b.jdk.incubator.vector.Double64Vector.scatter 0.879784798 
b.jdk.incubator.vector.Double256Vector.MULMaskedLanes 0.883884693
b.jdk.incubator.vector.IntScalar.rearrange064 0.894557645
b.jdk.incubator.vector.DoubleScalar.gather512 0.894795847

This patch didn't touch any code about these APIs. I suppose it maybe due to the unstable testing environment.

sandlerwang commented:
> 

These cases related to lane got 10%+ improvement. I will have a double check for the tests not related to your patches.
b.jdk.incubator.vector.Int64Vector.withLane	2.441469972
b.jdk.incubator.vector.LongMaxVector.withLane	2.418128287
b.jdk.incubator.vector.Long128Vector.withLane	2.413777238
b.jdk.incubator.vector.IntMaxVector.withLane	2.369558104
b.jdk.incubator.vector.Short64Vector.withLane	2.35949239
b.jdk.incubator.vector.Int128Vector.withLane	2.358443311
b.jdk.incubator.vector.Byte64Vector.withLane	2.243548282
b.jdk.incubator.vector.Short128Vector.withLane	2.229173881
b.jdk.incubator.vector.ShortMaxVector.withLane	2.228188826
b.jdk.incubator.vector.Double128Vector.withLane	2.076927965
b.jdk.incubator.vector.DoubleMaxVector.withLane	2.07473915
b.jdk.incubator.vector.Byte128Vector.withLane	2.066975989
b.jdk.incubator.vector.ByteMaxVector.withLane	2.059368462
b.jdk.incubator.vector.Int64Vector.MAXLanes	1.717211023
b.jdk.incubator.vector.Int64Vector.MINLanes	1.703460064
b.jdk.incubator.vector.Float64Vector.withLane	1.654619126
b.jdk.incubator.vector.Float128Vector.withLane	1.623629282
b.jdk.incubator.vector.FloatMaxVector.withLane	1.620377164
b.jdk.incubator.vector.Int64Vector.MAXMaskedLanes	1.618505915
b.jdk.incubator.vector.Int64Vector.ADDMaskedLanes	1.618226116
b.jdk.incubator.vector.Int64Vector.MINMaskedLanes	1.612555329
b.jdk.incubator.vector.Int64Vector.ADDLanes	1.572023834
b.jdk.incubator.vector.ByteMaxVector.laneextract	1.512898244
b.jdk.incubator.vector.Byte128Vector.laneextract	1.511894956
b.jdk.incubator.vector.Short64Vector.laneextract	1.501446026
b.jdk.incubator.vector.Byte64Vector.MAXLanes	1.496241406
b.jdk.incubator.vector.Byte64Vector.laneextract	1.478740319
b.jdk.incubator.vector.Short128Vector.laneextract	1.477009713
b.jdk.incubator.vector.ShortMaxVector.laneextract	1.476753991
b.jdk.incubator.vector.Byte64Vector.MINLanes	1.465747678
b.jdk.incubator.vector.Byte64Vector.MAXMaskedLanes	1.454729946
b.jdk.incubator.vector.Short64Vector.MAXMaskedLanes	1.428170142
b.jdk.incubator.vector.Short64Vector.MAXLanes	1.392436004
b.jdk.incubator.vector.Long128Vector.laneextract	1.385026499
b.jdk.incubator.vector.Short64Vector.MINLanes	1.384920334
b.jdk.incubator.vector.LongMaxVector.laneextract	1.382036611
b.jdk.incubator.vector.Int64Vector.laneextract	1.370389562
b.jdk.incubator.vector.Byte64Vector.MINMaskedLanes	1.368841973
b.jdk.incubator.vector.Short64Vector.MINMaskedLanes	1.356288446
b.jdk.incubator.vector.IntMaxVector.laneextract	1.320549168
b.jdk.incubator.vector.Int128Vector.laneextract	1.318487562
b.jdk.incubator.vector.Long512Vector.MAXMaskedLanes	1.194691611
b.jdk.incubator.vector.Short256Vector.MAXMaskedLanes	1.190682334
b.jdk.incubator.vector.Short256Vector.MULMaskedLanes	1.188007186
b.jdk.incubator.vector.DoubleMaxVector.laneextract	1.170442569
b.jdk.incubator.vector.Double128Vector.laneextract	1.169245297
b.jdk.incubator.vector.Float64Vector.laneextract	1.169160601
b.jdk.incubator.vector.Int512Vector.broadcast	1.154768618
b.jdk.incubator.vector.Byte64Vector.ADDMaskedLanes	1.150991683
b.jdk.incubator.vector.Float512Vector.MULMaskedLanes	1.148539242
b.jdk.incubator.vector.Byte128Vector.gatherMasked	1.135360038
b.jdk.incubator.vector.Double512Vector.SQRTMasked	1.12119304
b.jdk.incubator.vector.Float128Vector.laneextract	1.120281215
b.jdk.incubator.vector.FloatMaxVector.laneextract	1.118768235

Commit:
8281505: Add CompileCommand PrintIdealPhase

Reviewed-by: kvn, thartmann, chagedorn

rwestrel committed:
8280799: С2: assert(false) failed: cyclic dependency prevents range check elimination

Reviewed-by: thartmann, kvn

robcasloz committed:
8281539: IGV: schedule approximation computes immediate dominators wrongly

Replace custom dominator computation with one from the WALA libraries.

Reviewed-by: neliasso, chagedorn

kstefanj committed:
8281637: Remove unused VerifyOption_G1UseNextMarking

Reviewed-by: tschatzl, iwalulya

albertnetymk committed:
8280783: Parallel: Refactor PSCardTable::scavenge_contents_parallel

Reviewed-by: tschatzl, iwalulya

eme64 committed:
8281553: Ensure we only require liveness from mach-nodes with barriers

Reviewed-by: neliasso, chagedorn

magicus committed:
8274980: Improve adhoc build version strings

Reviewed-by: shade, erikj

Commit:
8256368: Avoid repeated upcalls into Java to re-resolve MH/VH linkers/invokers

Reviewed-by: dlong, kvn

lkorinth committed:
8281585: Remove unused imports under test/lib and jtreg/gc

Reviewed-by: dholmes, sspitsyn

brianjstafford committed:
8274238: Inconsistent type for young_list_target_length()

Reviewed-by: kbarrett, tschatzl

tstuefe committed:
8280940: gtest os.release_multi_mappings_vm is racy

Reviewed-by: dcubed, sjohanss

Commit:
8280473: CI: Support unresolved JVM_CONSTANT_Dynamic constant pool entries

Reviewed-by: dlong, redestad, neliasso

iklam committed:
8281678: appcds/dynamicArchive/ArchiveConsistency.java fails after JDK-8279997

Reviewed-by: shade, dcubed

prrace committed:
8269091: javax/sound/sampled/Clip/SetPositionHang.java failed with ArrayIndexOutOfBoundsException: Array index out of range: -4

Reviewed-by: serb

jaikiran committed:
8281634: jdeps: java.lang.InternalError: Missing message: err.invalid.filters

Reviewed-by: dfuchs, naoto, mchung

honkar-jdk committed:
8016524: [macosx] Bottom line is not visible for JTableHeader

Reviewed-by: psadhukhan, prr

shipilev committed:
8281467: Allow larger OptoLoopAlignment and CodeEntryAlignment

Reviewed-by: kvn, dlong

Commit:
8280916: Simplify HotSpot Style Guide editorial changes

Reviewed-by: dcubed, dholmes, stuefe, stefank, kvn, tschatzl

TobiHartmann committed:
8281713: [BACKOUT] AArch64: Implement string_compare intrinsic in SVE

Reviewed-by: kvn, dlong

Commit:
8281728: Redundant null check in LineNumberInputStream.read

Reviewed-by: redestad

Commit:
8281722: Removal of PrintIdealLevel

Reviewed-by: chagedorn, thartmann

pashh committed:
8280415: Remove EA from JDK 18 version string starting with Initial RC promotion B35 on February 10, 2022

Reviewed-by: erikj, iris

dmarkov20 committed:
8281555: [macos] Get rid of deprecated Style Masks constants

Reviewed-by: serb, aivanov

Commit:
8281741: [testbug] PrintIdealPhaseTest fails with -Xcomp

Reviewed-by: kvn, chagedorn, thartmann

magicus committed:
8246033: bin/print_config.js script uses nashorn jjs tool

Reviewed-by: erikj

albertnetymk committed:
8280136: Serial: Remove unnecessary use of ExpandHeap_lock

Reviewed-by: iwalulya, kbarrett, sjohanss

kstefanj committed:
8281748: runtime/logging/RedefineClasses.java failed "assert(addr != __null) failed: invariant"

Reviewed-by: ayang, kbarrett

shipilev committed:
8281744: x86: Use short jumps in TIG::set_vtos_entry_points

Reviewed-by: rehn, coleenp

calvinccheung committed:
8281186: runtime/cds/appcds/DumpingWithNoCoops.java fails

Reviewed-by: minqi, iklam, stuefe

sormuras committed:
8280825: Modules that "provide" ToolProvider should document the name that can be used

Reviewed-by: jjg, lancea, alanb

lmesnik committed:
8279949: JavaThread::_free_handle_block leaks native memory

Reviewed-by: dholmes, coleenp

lmesnik committed:
8281771: Crash in java_lang_invoke_MethodType::print_signature

Reviewed-by: dholmes, shade

merykitty committed:
8278173: [vectorapi] Add x64 intrinsics for unsigned (zero extended) casts

Reviewed-by: psandoz, sviswanathan

Commit:
8236136: tests which use CompilationMode shouldn't be run w/ TieredStopAtLevel

Reviewed-by: neliasso, kvn, thartmann

shipilev committed:
8281812: x86: Use short jumps in TemplateTable::condy_helper

Reviewed-by: redestad, neliasso

DamonFool committed:
8281829: runtime/CommandLine/OptionsValidation/TestOptionsWithRanges.java fails after JDK-8281467

Reviewed-by: kvn, thartmann

egahlin committed:
8281948: JFR: Parser skips too many bytes for fractional types

Reviewed-by: mgronlun

Michael-Mc-Mahon committed:
8278067: Make HttpURLConnection default keep alive timeout configurable

Reviewed-by: dfuchs

eme64 committed:
8281732: add assert for non-NULL assumption for return of unique_ctrl_out

Reviewed-by: kvn, chagedorn, thartmann

mahendrachhipa committed:
8061729: Update java/net tests to eliminate dependency on sun.net.www.MessageHeader and some other internal APIs

Reviewed-by: dfuchs

naotoj committed:
8176706: Additional Date-Time Formats

Reviewed-by: joehw, rriggs

Commit:
8281170: Test jdk/tools/jpackage/windows/WinInstallerIconTest always fails on Windows 11

Reviewed-by: almatvee

Commit:
8281874: Can't unpack msi installers from test/jdk/tools/jpackage/windows/test/jdk/tools/jpackage/windows/WinShortcutPromptTest.java test

Reviewed-by: almatvee

Commit:
8281335: Allow a library already loaded via System::loadLibrary to be loaded as a raw library

Reviewed-by: sundar, mcimadamore

shipilev committed:
8281822: Test failures on non-DTrace builds due to incomplete DTrace* flags handling

Reviewed-by: dholmes, kvn

tprinzing committed:
8281003: MethodHandles::lookup throws NPE if caller is null

Reviewed-by: ihse, mchung, jrose, alanb

dfuch committed:
8282020: ProblemList sun/net/www/protocol/https/HttpsURLConnection/B6216082.java until JDK-8282017 is fixed

Reviewed-by: michaelm, naoto

desruisseaux committed:
8275345: RasterFormatException when drawing a tiled image made of non-writable rasters

Reviewed-by: prr, aivanov

jddarcy committed:
8281671: Class.getCanonicalName spec should explicitly cover array classes

Reviewed-by: mchung

Commit:
8282011: test/jdk/tools/jpackage/windows/WinL10nTest.java test fails if light.exe is not in %PATH%

Reviewed-by: almatvee

JesperIRL committed:
Merge

Commit:
8282007: Assorted enhancements to jpackage testing framework

Reviewed-by: almatvee

Commit:
8281803: AArch64: Optimize masked vector NOT/AND_NOT for SVE

Reviewed-by: aph, njian

iklam committed:
8279969: NULL return from map_bitmap_region() needs to be checked

Reviewed-by: ccheung, coleenp

backwaterred committed:
8203290: [AIX] Check functionality of JDK-8199712 (Flight Recorder)

Implements JFR for AIX

Reviewed-by: erikj, mdoerr, mgronlun, stuefe, ihse

prsadhuk committed:
8264743: Add forRemoval for deprecated classes and method in javax/swing/plaf/basic

Reviewed-by: trebari, prr

Commit:
8281305: Test com/sun/net/httpserver/simpleserver/MapToPathTest.java fails on Windows 11

Reviewed-by: dfuchs

albertnetymk committed:
8281971: Remove unimplemented InstanceRefKlass::do_next

Reviewed-by: dholmes

Commit:
8282019: Unused static fields DEGREES_TO_RADIANS, RADIANS_TO_DEGREES in StrictMath

Reviewed-by: bpb, darcy

jddarcy committed:
8268250: Class.arrayType() for a 255-d array throws undocumented IllegalArgumentException

Reviewed-by: sundar, alanb

tprinzing committed:
8281000: ClassLoader::registerAsParallelCapable throws NPE if caller is null

Reviewed-by: erikj, ihse, mchung, bchristi

mahendrachhipa committed:
8282017: sun/net/www/protocol/https/HttpsURLConnection/B6216082.java fails with "SocketException: Unexpected end of file from server"

Reviewed-by: dfuchs

naotoj committed:
8281317: CompactNumberFormat displays 4-digit values when rounding to a new range

Reviewed-by: joehw

Commit:
8282075: ProblemList 3 compiler/whitebox tests on macosx-x64

Reviewed-by: mikael, bpb

magicus committed:
8188073: Add Capstone as backend for hsdis

Co-authored-by: Magnus Ihse Bursie <ihse@openjdk.org>
Co-authored-by: Jorn Vernee <jvernee@openjdk.org>
Reviewed-by: erikj

DamonFool committed:
8282025: assert(ctrl != __null) failed: control out is assumed to be unique after JDK-8281732

Reviewed-by: kvn, thartmann, chagedorn

prsadhuk committed:
8037573: Typo in DefaultTreeModel docs: askAllowsChildren instead of asksAllowsChildren

Reviewed-by: prr, jdv, azvegint

DamonFool committed:
8281936: compiler/arguments/TestCodeEntryAlignment.java fails on AVX512 machines

Reviewed-by: shade, kvn

robcasloz committed:
8279068: IGV: Update to work with JDK 16 and 17

Reviewed-by: kvn, neliasso, chagedorn

Commit:
8281267: VM HeapDumper dumps array classes several times

Reviewed-by: cjplummer, coleenp

lahodaj committed:
8277300: Issues with javadoc support for preview features

Reviewed-by: prappo, jjg

albertnetymk committed:
8282089: [BACKOUT] Parallel: Refactor PSCardTable::scavenge_contents_parallel

Reviewed-by: tschatzl

lahodaj committed:
8280866: SuppressWarnings does not work properly in package-info and module-info

Reviewed-by: darcy, vromero

pavelrappo committed:
8282056: Clean up com.sun.tools.javac.util.GraphUtils

Reviewed-by: jjg, mcimadamore, vromero

magicus committed:
8282086: Update jib profile to not set build to 0

Reviewed-by: erikj

Commit:
8282049: AArch64: Use ZR for integer zero immediate volatile stores

Reviewed-by: adinn, phh

Commit:
8282103: fix macosx-generic typo in ProblemList

Reviewed-by: rriggs

wangweij committed:
8255266: Update Public Suffix List to 3c213aa

Reviewed-by: xuelei

Commit:
8282044: [JVMCI] Export _sha3_implCompress, _md5_implCompress and aarch64::_has_negatives stubs to JVMCI compiler.

Reviewed-by: kvn

Commit:
8277488: Add expiry exception for Digicert (geotrustglobalca) expiring in May 2022

Reviewed-by: weijun

slumericanB committed:
8282077: PKCS11 provider C_sign() impl should handle CKR_BUFFER_TOO_SMALL error

Reviewed-by: mikael

shipilev committed:
8281815: x86: Use short jumps in TIG::generate_slow_signature_handler

Reviewed-by: rrich, dholmes, jiefu

TobiHartmann committed:
8282085: The REGISTER_DEFINITION macro is useless after JDK-8269122

Reviewed-by: jiefu, chagedorn, kvn

eme64 committed:
8281544: assert(VM_Version::supports_avx512bw()) failed for Tests jdk/incubator/vector/

Reviewed-by: kvn, neliasso, thartmann

Commit:
8282158: ECParameters InvalidParameterSpecException messages missed ECKeySizeParameterSpec

Reviewed-by: xuelei

albertnetymk committed:
8282096: G1: Remove redundant checks in G1CardSet::free_mem_object

Reviewed-by: tschatzl, iwalulya

Commit:
8282166: JDK-8282158 changed ECParameters' package by accident

Reviewed-by: xuelei

Commit:
8282130: (bf) Remove unused ARRAY_BASE_OFFSET, ARRAY_INDEX_SCALE from read-only Heap Buffers

Reviewed-by: bpb, alanb

manukumarvs committed:
8282147: [TESTBUG] waitForIdle after creating frame in JSpinnerMouseAndKeyPressTest.java

Reviewed-by: aivanov

magicus committed:
8253757: Add LLVM-based backend for hsdis

Co-authored-by: Magnus Ihse Bursie <ihse@openjdk.org>
Co-authored-by: Ludovic Henry <luhenry@openjdk.org>
Co-authored-by: Jorn Vernee <jvernee@openjdk.org>
Co-authored-by: Nick Gasson <ngasson@openjdk.org>
Reviewed-by: erikj, luhenry

mkartashev committed:
8280861: Robot color picker broken on Linux with scaling above 100%

Reviewed-by: serb

djelinski committed:
8281523: Accessibility: Conversion from string literal loses const qualifier

Reviewed-by: prr, aivanov, kizune

jaikiran committed:
8282190: Typo in javadoc of java.time.format.DateTimeFormatter#getDecimalStyle

Reviewed-by: dfuchs, rriggs, lancea, iris

manukumarvs committed:
8281745: Create a regression test for JDK-4514331

Reviewed-by: serb

eme64 committed:
8281543: Remove unused code/headerfile dtraceAttacher.hpp

Reviewed-by: thartmann

Commit:
8282220: contentType should not be a PKCS7's member

Reviewed-by: xuelei

aivanov-jdk committed:
8260328: Drop redundant CSS properties from java.desktop HTML files

Reviewed-by: serb, dmarkov

coleenp committed:
8271008: appcds/*/MethodHandlesAsCollectorTest.java tests time out because of excessive GC (CodeCache GC Threshold) in loom

Reviewed-by: thartmann, eosterlund

Commit:
8276686: Malformed Javadoc inline tags in JDK source in /java/util/regex/Pattern.java

Reviewed-by: iris, bpb, lancea

magicus committed:
8244593: Clean up GNM/NM after JEP 381

Reviewed-by: erikj

Commit:
8280958: G1/Parallel: Unify marking code structure

Reviewed-by: ayang, kbarrett

Commit:
8281315: Unicode, (?i) flag and backreference throwing IndexOutOfBounds Exception

Reviewed-by: naoto

backwaterred committed:
8282042: [testbug] FileEncodingTest.java depends on default encoding

Adds expected encoding "ISO-8859-1" for AIX in FileEncodingTest.java

Reviewed-by: naoto

liach committed:
8261407: ReflectionFactory.checkInitted() is not thread-safe

Co-authored-by: Peter Levart <plevart@openjdk.org>
Reviewed-by: dholmes, mchung, plevart

Commit:
5041655: (ch) FileLock: negative param and overflow issues

Reviewed-by: alanb

prrace committed:
8282276: Problem list failing two Robot Screen Capture tests

Reviewed-by: dcubed

prrace committed:
8280964: [Linux aarch64] : drawImage dithers TYPE_BYTE_INDEXED images incorrectly

Reviewed-by: serb, dmarkov

zhengyu123 committed:
8281615: Deadlock caused by jdwp agent

Reviewed-by: dholmes, cjplummer

simonis committed:
8281962: Avoid unnecessary native calls in InflaterInputStream

Reviewed-by: clanger, redestad, alanb, lancea

Commit:
8282231: x86-32: runtime call to SharedRuntime::ldiv corrupts registers

Reviewed-by: shade, jiefu

Commit:
8282194: C1: Missing side effects of dynamic constant linkage

Reviewed-by: kvn, thartmann

Commit:
8255577: Possible issues with SR_initialize

Reviewed-by: shade, stuefe

shipilev committed:
8282225: GHA: Allow one concurrent run per PR only

Reviewed-by: ihse

Commit:
8282208: Reduce MachNode size

Reviewed-by: kvn, thartmann, jiefu

XueleiFan committed:
8282279: Interpret case-insensitive string locale independently

Reviewed-by: weijun

pavelrappo committed:
8281376: Consider polymorphic methods when looking for overrides

Reviewed-by: hannesw

jonathan-gibbons committed:
8281217: Source file launch with security manager enabled fails

Reviewed-by: sundar

Commit:
8280409: JarFile::getInputStream can fail with NPE accessing ze.getName()

Reviewed-by: mullan, alanb

XueleiFan committed:
8282309: Operation before upper case conversion

Reviewed-by: valeriep, wetmore

Michael-Mc-Mahon committed:
8282296: (se) Pipe.open() creates a Pipe implementation that uses Unix domain sockets (win)

Reviewed-by: dfuchs, lancea, bpb

Commit:
8280901: MethodHandle::linkToNative stub is missing w/ -Xint

Reviewed-by: shade, kvn

plummercj committed:
8282076: Merge some debug agent changes from the loom repo

Reviewed-by: amenkov, lmesnik

manukumarvs committed:
8281988: Create a regression test for JDK-4618767

Reviewed-by: aivanov

plummercj committed:
8281614: serviceability/sa/ClhsdbFindPC.java fails with java.lang.RuntimeException: 'In code in NMethod for jdk/test/lib/apps/LingeredApp.steadyState' missing from stdout/stderr

Reviewed-by: dcubed, dholmes

albertnetymk committed:
8282299: Remove unused PartialArrayScanTask default constructor

Reviewed-by: tschatzl

prsadhuk committed:
8277369: Strange behavior of JMenuBar with RIGHT_TO_LEFT orientation, arrow keys behaves opposite traversing through keyboard

Reviewed-by: prr, naoto, serb

Commit:
AArch64: Optimize out WHILELO with PTRUE

This patch uses PTRUE instruction to create predicate registers for
partial vector operations which would be efficient than WHILELO
instruction according to the software optimization guide of N2[1] and
V1[2].

[1] https://developer.arm.com/documentation/PJDOC-466751330-18256/0001
[2] https://developer.arm.com/documentation/pjdoc466751330-9685/latest/

Change-Id: I9be5aa82ab567e19c62c698dc9c4d852efd2f607

Commit:
8282188: Unused static field MathContext.DEFAULT_DIGITS

Reviewed-by: darcy, bpb

coleenp committed:
8262400: runtime/exceptionMsgs/AbstractMethodError/AbstractMethodErrorTest.java fails in test_ame5_compiled_vtable_stub with wrapper

Reviewed-by: dholmes, lmesnik

parttimenerd committed:
8282200: ShouldNotReachHere() reached by AsyncGetCallTrace after JDK-8280422

Reviewed-by: dholmes, mdoerr, kevinw

jonathan-gibbons committed:
8281944: JavaDoc throws java.lang.IllegalStateException: ERRONEOUS

Reviewed-by: hannesw

XueleiFan committed:
8282316: Operation before String case conversion

Reviewed-by: valeriep

a74nh committed:
8277204: Implement PAC-RET branch protection on Linux/AArch64

Reviewed-by: erikj, ihse, adinn, ngasson

albertnetymk committed:
8282348: Remove unused CardTable::dirty_card_iterate

Reviewed-by: kbarrett

albertnetymk committed:
8282307: Parallel: Incorrect discovery mode in PCReferenceProcessor

Reviewed-by: tschatzl, kbarrett

djelinski committed:
8281525: Enable Zc:strictStrings flag in Visual Studio build

Reviewed-by: dholmes, ihse

Commit:
8005885: enhance PrintCodeCache to print more data

Reviewed-by: xliu, phh

Commit:
8280357: user.home = "?" when running with systemd DynamicUser=true

Reviewed-by: naoto, alanb

kevinjwalls committed:
8206187: javax/management/remote/mandatory/connection/DefaultAgentFilterTest.java fails with Port already in use

Reviewed-by: msheppar, amenkov

cl4es committed:
8282347: AARCH64: Untaken branch in has_negatives stub

Reviewed-by: aph, haosun, thartmann

tkiriyama committed:
8280684: JfrRecorderService failes with guarantee(num_written > 0) when no space left on device.

Reviewed-by: mgronlun

tstuefe committed:
8281015: Further simplify NMT backend

Reviewed-by: zgu, mbaesken

MBaesken committed:
8282345: handle latest VS2022 in abstract_vm_version

Reviewed-by: dholmes, mdoerr

egahlin committed:
8271232: JFR: Scrub recording data

Reviewed-by: mgronlun

egahlin committed:
8281739: JFR: Use message with Objects.requireNonNull

Reviewed-by: mgronlun

Commit:
8279995: jpackage --add-launcher option should allow overriding description

Reviewed-by: asemenyuk

Commit:
8282219: jdk/java/lang/ProcessBuilder/Basic.java fails on AIX

Reviewed-by: rriggs

egahlin committed:
8282153: JFR: Check for recording waste

Reviewed-by: mgronlun

Commit:
8282398: EndingDotHostname.java test fails because SSL cert expired

Reviewed-by: xuelei

Commit:
8282428: ProblemList jdk/jfr/jvm/TestWaste.java

Reviewed-by: mikael

shipilev committed:
8281507: Two javac tests have bad jtreg `@clean` tags

Reviewed-by: darcy

JohnTortugo committed:
8267265: Use new IR Test Framework to create tests for C2 Ideal transformations

Reviewed-by: chagedorn

eme64 committed:
8230382: Clean up ConvI2L, CastII and CastLL::Ideal methods

Reviewed-by: thartmann, roland

Commit:
8282360: Merge POSIX implementations of ThreadCritical

Reviewed-by: stuefe, coleenp, dholmes

naotoj committed:
8282131: java.time.ZoneId should be a sealed abstract class

Reviewed-by: iris, rriggs, bpb, lancea, mchung, scolebourne

shipilev committed:
8282172: CompileBroker::log_metaspace_failure is called from non-Java/compiler threads

Reviewed-by: kvn, thartmann

wangweij committed:
8277976: Break up SEQUENCE in X509Certificate::getSubjectAlternativeNames and X509Certificate::getIssuerAlternativeNames in otherName
6776681: Invalid encoding of an OtherName in X509Certificate.getAlternativeNames()

Reviewed-by: mullan

coleenp committed:
8282240: Add _name field to Method for NOT_PRODUCT only

Reviewed-by: pchilanomate, hseigel

iklam committed:
8275731: CDS archived enums objects are recreated at runtime

Reviewed-by: coleenp, ccheung

jddarcy committed:
8282462: Remove unnecessary use of @SuppressWarnings("preview")

Reviewed-by: vromero

jddarcy committed:
8282464: Remove author tags from java.compiler

Reviewed-by: jjg

Commit:
8281210: Add manpage changes for PAC-RET protection on Linux/AArch64

Reviewed-by: iklam

a74nh committed:
8282392: [zero] Build broken on AArch64

Reviewed-by: aph, shade
